src/app/preferences/index.tsx

```diff
@@ -0,0 +1,107 @@
+
+interface PreferencesDialogProps {
+  haveParamsChanged;
+  setHaveParamsChanged;
+}
+
+interface Section {
+  section: string;
+  subsection: string;
+  title: string;
+}
+
+type Sections = Section[];
+
+const PreferencesDialog = (props: PreferencesDialogProps) => {
+  const { haveParamsChanged, setHaveParamsChanged } = props;
+  const { translate } = useTranslationContext();
+  const dispatch = useAppDispatch();
+  const SECTIONS: Sections = [
+    { section: 'general', subsection: 'general', title: translate('preferences.general.title') },
+    { section: 'workspace', subsection: 'overview', title: translate('preferences.workspace.overview.title') },
+    { section: 'workspace', subsection: 'members', title: translate('preferences.workspace.members.title') },
+    { section: 'workspace', subsection: 'teams', title: translate('preferences.workspace.teams.title') },
+    { section: 'workspace', subsection: 'billing', title: translate('preferences.workspace.billing.title') },
+    { section: 'account', subsection: 'plans', title: translate('preferences.account.plans.title') },
+    { section: 'account', subsection: 'billing', title: translate('preferences.account.billing.title') },
+    { section: 'account', subsection: 'security', title: translate('preferences.account.security.title') },
+  ];
+
+  const params = new URLSearchParams(window.location.search);
+  const currentSectionParams = params.getAll('section');
+  const currentSubsectionParams = params.getAll('subsection');
+
+  const [activeSection, setActiveSection] = useState<Sections>([{ section: '', subsection: '', title: '' }]);
+
+  useEffect(() => {
+    if (currentSectionParams.length > 0) {
+      const sectionParams = SECTIONS.filter((section) => section.section === currentSectionParams[0]);
+      const subsectionParams = sectionParams.filter(
+        (subsection) => subsection.subsection === currentSubsectionParams[0],
+      );
+      changeSection(sectionParams[0].section, subsectionParams[0].subsection);
+      setHaveParamsChanged(false);
+    } else {
+      dispatch(uiActions.setIsPreferencesDialogOpen(false));
+      setHaveParamsChanged(false);
+    }
+  }, [haveParamsChanged]);
+
+  const changeSection = (onSection, OnSubsection) => {
+    const selectedSection = SECTIONS.filter((section) => section.section === onSection);
+    const selectedSubsection = selectedSection.filter((subsection) => subsection.subsection === OnSubsection);
+    setActiveSection(selectedSubsection);
+  };
+
+  const goSection = (section, subsection) => {
+    navigationService.openPreferencesDialog(section, subsection);
+    changeSection(section, subsection);
+  };
+
+  const onClose = () => {
+    dispatch(uiActions.setIsPreferencesDialogOpen(false));
+    navigationService.closePreferencesDialog();
+  };
+
+  return (
+    <Modal maxWidth="w-full max-w-4xl" className="m-0 flex" isOpen={true} onClose={() => onClose()}>
```

Comment:
to display the modal could pass the `isOpen` prop from the parent

Reply:
Done